promgen/static/js/promgen.vue.js

```diff
@@ -180,7 +197,16 @@ app.component('silence-form', {
                     }
                 });
         },
-    }
+        hideModal: function (modalId) {
+            var modal = $('#' + modalId);
+            if (modal.length) {
+                globalStore.setMessages([]);
+                this.form = {};
+                this.state = silenceStore.state;
+                modal.modal('hide');
+            }
+        },
+    },
```

Comment:
```suggestion
    },
    watch: {
        "state.show"(val) {
            if (val) {
                this.showModal();
            } else {
                this.hideModal();
            }
        },
    },
```

This watcher inside the `silence-modal` component will react on changes to the `show` variable from the silence store.

Since external components cannot run the `showModal` or `hideModal` methods from the `silence-modal` component, we trigger those functions when the state changes. And the state can be changed from anywhere via the silence store.